include/evmc/evmc.h

```diff
@@ -685,6 +685,26 @@ typedef struct evmc_result (*evmc_execute_fn)(struct evmc_instance* instance,
                                               size_t code_size);
 
 
+struct evmc_tracer_context;
+
+typedef void (*evmc_trace_callback)(struct evmc_tracer_context* context,
+                                    int depth,
+                                    int step,
+                                    size_t code_offset,
+                                    enum evmc_status_code status_code,
+                                    int64_t gas_left,
+                                    size_t stack_num_items,
+                                    const struct evmc_uint256be* pushed_stack_item,
+                                    size_t memory_size,
+                                    size_t changed_memory_offset,
+                                    size_t changed_memory_size,
+                                    const uint8_t* changed_memory);
+
+typedef void (*evmc_set_tracer)(struct evmc_instance* instance,
```

Comment:
Shouldn't this return a pointer to previously set callback?
Why do you need to declare a pointer to function (`evmc_set_tracer`) if it's just to declare one function?

Reply:
Yes, it can return previous callback, but it would be hard to also return the context, so you will not be able to restore previous state anyway.

(If I understand the question) The evmc_instance is a set of function pointers. It is easier to read the spec when function pointer types are declared separately. I forgot the `_fn` suffix in this type name.

Reply:
Ah ok, I didn't notice that it's a part of `evmc_instance`, sorry